asapdiscovery-ml/asapdiscovery/ml/inference/inference.py

```diff
@@ -173,8 +176,39 @@ def predict(self, g: dgl.DGLGraph):
         g : dgl.DGLGraph
             DGLGraph object.
 
+        Returns
+        -------
+        np.ndarray
+            Predictions for each graph.
         """
         with torch.no_grad():
             output_tensor = self.model(g, g.ndata["h"])
+            # unsure exactly why this reshape is done
             output_tensor = torch.reshape(output_tensor, (-1, 1))
-            return output_tensor.cpu().numpy()
+            # we ravel it away to get a 1D array
+            return output_tensor.cpu().numpy().ravel()
+
+    def predict_from_smiles(
+        self, smiles: Union[str, list[str]], **kwargs
+    ) -> np.ndarray:
+        """Predict on a list of SMILES strings, or a single SMILES string.
+
+        Parameters
+        ----------
+        smiles : Union[str, List[str]]
+            SMILES string or list of SMILES strings.
+
+        Returns
+        -------
+        np.ndarray
+            Predictions for each SMILES string.
+        """
+        if isinstance(smiles, str):
+            smiles = [smiles]
+
+        gids = GraphInferenceDataset(
+            smiles, node_featurizer=CanonicalAtomFeaturizer(), **kwargs
+        )
+
+        data = [self.predict(g) for g in gids]
+        return np.concatenate(np.asarray(data))
```

Comment:
in the case of a single SMILES, does the predicted value get returned as `value` or as `[value]`? I think returning as a float would make more sense?

Reply:
Yeah its being returned as [value], I can change to returning `value` for a single smiles.